server/src/main/java/com/defold/extender/Extender.java

```diff
@@ -478,66 +513,144 @@ private File compileMain(File maincpp, Map<String, Object> manifestContext) thro
         return getExtensionManifestFiles().stream().map(File::getParentFile).collect(Collectors.toList());
     }
 
-    private void buildExtension(File manifest, Map<String, Object> manifestContext) throws IOException, InterruptedException, ExtenderException {
-        File extDir = manifest.getParentFile();
-        File srcDir = new File(extDir, "src");
+    private List<File> listFiles(File[] srcDirs, String regEx) {
+        List<File> srcFiles = new ArrayList<>();
+        for (File srcDir : srcDirs) {
+            if (srcDir.exists() && srcDir.isDirectory()) {
+                List<File> _srcFiles =  new ArrayList<>(FileUtils.listFiles(srcDir, null, true));
+                _srcFiles = Extender.filterFiles(_srcFiles, regEx);
+
+                // sorting makes it easier to diff different builds
+                Collections.sort(_srcFiles, NameFileComparator.NAME_INSENSITIVE_COMPARATOR);
+                srcFiles.addAll(_srcFiles);
+            }
+        }
+        return srcFiles;
+    }
+
+    private List<File> listFiles(File srcDir, String regEx) {
+        File[] srcDirs = {srcDir};
+        return listFiles(srcDirs, regEx);
+    }
 
-        File[] srcFiles = null;
-        if (srcDir.isDirectory()) {
-            Collection<File> _srcFiles = new ArrayList<>();
-            _srcFiles = FileUtils.listFiles(srcDir, null, true);
-            _srcFiles = Extender.filterFiles(_srcFiles, platformConfig.sourceRe);
+    private List<File> generateProtoCxxForEngine(File extDir, Map<String, Object> manifestContext, List<File> protoFiles) throws IOException, InterruptedException, ExtenderException {
```

Comment:
This function converts .proto files into _ddf.h/_ddf.cpp, that are used by the engine to read the files at runtime.